v3/src/components/axis/models/multi-scale.ts

```diff
@@ -121,4 +122,29 @@ export class MultiScale {
     return {data: NaN}
   }
 
+  formatValueForScale(value: number) {
+
+    function formatNumber(n: number, dom: [number, number], range: [number, number]): string {
```

Comment:
FYI, you could define a type for this (or even two types, AxisDomain and AxisRange, in case that was helpful):
```suggestion
    export type AxisExtent = [number, number] // might need to move this outside the class definition
    
    function formatNumber(n: number, domain: AxisExtent, range: AxisExtent): string {
```